src/librbd/io/AioCompletion.cc

```diff
@@ -34,10 +34,8 @@ namespace io {
 int AioCompletion::wait_for_complete() {
   tracepoint(librbd, aio_wait_for_complete_enter, this);
   {
-    std::unique_lock<std::mutex> locker(lock);
-    while (state != AIO_STATE_COMPLETE) {
-      cond.wait(locker);
-    }
+    waiting_for_complete.store(true, std::memory_order_release);
+    done.wait(false, std::memory_order_acquire);
   }
```

Comment:
These braces were defining the critical section of the mutex.  They are useless now -- let's remove them and reindent.